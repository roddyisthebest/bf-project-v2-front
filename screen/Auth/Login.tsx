import React from 'react';
import {Pressable, Text} from 'react-native';
import styled from 'styled-components/native';

const Container = styled.View`
  align-items: center;
  flex: 1;
  padding: 0 30px;
`;

const TitleText = styled.Text`
  color: black;
  font-weight: 900;
  font-size: 27px;
  margin: 35px 0 50px 0;
`;

const LoginBtn = styled.TouchableOpacity`
  width: 100%;
  height: 70px;
  border: 1px solid #cdcdcd;
  border-radius: 100px;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;

const LoginText = styled.Text`
  color: black;
  font-weight: 800;
  font-size: 25px;
  margin-left: 15px;
`;
const LoginBkg = styled.Image`
  width: 25px;
  height: 25px;
`;

const Login = ({
  navigation: {navigate},
}: {
  navigation: {navigate: Function};
}) => {
  return (
    <Container>
      <TitleText>
        큰숲 청년부에서 무슨 일이 {'\n'}일어나고 있는지 알아보세요.
      </TitleText>
      <LoginBtn
        onPress={() => {
          navigate('Auth', {
            screen: 'SnsLogin',
          });
        }}>
        <LoginBkg source={require('../../assets/img/Kakao.png')} />
        <LoginText>KAKAO 로그인</LoginText>
      </LoginBtn>
      <Pressable
        onPress={() => {
          navigate('Auth', {
            screen: 'Code',
          });
        }}>
        <Text>code</Text>
      </Pressable>
    </Container>
  );
};

export default Login;
