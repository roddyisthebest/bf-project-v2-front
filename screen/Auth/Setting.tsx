import React, {useCallback, useEffect, useRef, useState} from 'react';
import styled from 'styled-components/native';
import {Switch} from 'react-native';

const Container = styled.View`
  flex: 1;
  align-items: center;
`;

const Title = styled.Text`
  font-size: 25px;
  font-weight: 800;
  color: black;
  margin-top: 20px;
`;

const SubTitle = styled.Text`
  font-size: 15px;
  font-weight: 400;
  color: #687684;
`;

const SwitchWrapper = styled.View`
  padding: 25px 40px;
  width: 100%;
  border-bottom-width: 0.5px;
  border-bottom-color: #ced5dc;
`;

const SwitchColumn = styled.View`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 10px;
`;

const SwitchTitle = styled.Text`
  color: black;
  font-size: 20px;
  font-weight: 800;
`;

const SwitchSubTitle = styled.Text`
  color: #687684;
  font-size: 12px;
  font-weight: 400;
`;

const Btn = styled.Pressable<{bkgColor: string}>`
  width: 95px;
  height: 35px;
  align-items: center;
  justify-content: center;
  border-radius: 40px;
  background-color: ${props => props.bkgColor};
  margin-top: 30px;
`;

const BtnText = styled.Text<{color: string}>`
  font-size: 15px;
  color: ${props => props.color};
  font-weight: 700;
`;
const Code = ({navigation: {dispatch}}: {navigation: {dispatch: Function}}) => {
  const [val, setVal] = useState([false, false, false]);
  const toggleVal = useCallback(
    (index: number) => {
      val.splice(index, 1, !val[index]);
      setVal([...val]);
    },
    [val],
  );
  return (
    <Container>
      <Title>기능 사용 설정</Title>
      <SubTitle style={{marginTop: 20}}>
        Welcome to roddyisthebest's world 🔥
      </SubTitle>
      <SubTitle>Please select the function you want to use.</SubTitle>
      <SwitchWrapper
        style={{borderTopColor: '#ced5dc', borderTopWidth: 0.5, marginTop: 25}}>
        <SwitchColumn>
          <SwitchTitle>매일성경</SwitchTitle>
          <Switch
            value={val[0]}
            onValueChange={() => {
              toggleVal(0);
            }}
          />
        </SwitchColumn>
        <SwitchSubTitle>
          매일성경(큐티) 업로드를 사용할 수 있습니다.
        </SwitchSubTitle>
      </SwitchWrapper>
      <SwitchWrapper>
        <SwitchColumn>
          <SwitchTitle>기도제목</SwitchTitle>
          <Switch
            value={val[1]}
            onValueChange={() => {
              toggleVal(1);
            }}
          />
        </SwitchColumn>
        <SwitchSubTitle>
          기도제목을 업로드하고 수정 및 삭제 할 수 있습니다.
        </SwitchSubTitle>
      </SwitchWrapper>
      <SwitchWrapper>
        <SwitchColumn>
          <SwitchTitle>벌금</SwitchTitle>
          <Switch
            value={val[2]}
            onValueChange={() => {
              toggleVal(2);
            }}
          />
        </SwitchColumn>
        <SwitchSubTitle>
          매일 성경(큐티) 벌금 로직에 참여할 수 있습니다.
        </SwitchSubTitle>
      </SwitchWrapper>
      <Btn bkgColor={'#10DDC2'}>
        <BtnText color={'white'}>확인</BtnText>
      </Btn>
    </Container>
  );
};

export default Code;
