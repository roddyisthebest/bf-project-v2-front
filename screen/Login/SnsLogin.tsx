import React, {useEffect, useState} from 'react';
import {View} from 'react-native';
import EncryptedStorage from 'react-native-encrypted-storage';
import {WebView} from 'react-native-webview';
import {useDispatch} from 'react-redux';
import {setToken} from '../../api';
import {login, setAuth, setUserInfo} from '../../store/slice';
import Config from 'react-native-config';
import {getMyInfo} from '../../api/user';
import {User} from '../../types/User';

const SnsLogin = () => {
  const dispatch = useDispatch();

  const IJECTED_JAVASCRIPT = `
  
  const element = document.getElementsByTagName("pre");
  
  if(element){
    window.ReactNativeWebView.postMessage(element[0].innerHTML);
    const body = document.getElementsByTagName("body");
    body[0].innerHTML= '잠시만 기다려주세요.'
  }
  `;

  const [accessToken, setAccessToken] = useState<string>();
  const [refreshToken, setRefreshToken] = useState<string>();
  const onWebViewMessage = (e: any) => {
    const jsonData = JSON.parse(e.nativeEvent.data);
    if (
      jsonData.payload.token.access_token &&
      jsonData.payload.token.refresh_token
    ) {
      setAccessToken(jsonData.payload.token.access_token);
      setRefreshToken(jsonData.payload.token.refresh_token);
    }
  };

  const setTokenInfo = async (access: string, refresh: string) => {
    await EncryptedStorage.setItem('refreshToken', refresh);
    await EncryptedStorage.setItem('accessToken', access);
    console.log('refresh', refresh);
    console.log('access', access);

    await setToken();
    dispatch(login(true));
    const {
      data: {payload},
    }: {data: {payload: User}} = await getMyInfo();
    dispatch(setUserInfo(payload));
    dispatch(setAuth(true));
  };

  useEffect(() => {
    if (accessToken && refreshToken) {
      setTokenInfo(accessToken, refreshToken);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [accessToken, refreshToken]);
  return (
    <View style={{flex: 1}}>
      <WebView
        userAgent="Chrome"
        source={{
          uri: `https://kauth.kakao.com/oauth/authorize?client_id=${Config.CLIENT_ID}&redirect_uri=${Config.KAKAO_CALLBACK}&response_type=code`,
        }}
        sharedCookiesEnabled={true}
        thirdPartyCookiesEnabled={true}
        javaScriptEnabled={true}
        injectedJavaScript={IJECTED_JAVASCRIPT}
        onMessage={onWebViewMessage}
        originWhitelist={['*']}
      />
    </View>
  );
};

export default SnsLogin;
