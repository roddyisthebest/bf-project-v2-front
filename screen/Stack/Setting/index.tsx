import React from 'react';
import {View, Text} from 'react-native';
const Setting = () => {
  return (
    <View style={{flex: 1}}>
      <Text>Setting</Text>
    </View>
  );
};

export default Setting;
