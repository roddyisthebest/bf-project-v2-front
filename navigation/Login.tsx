import {createNativeStackNavigator} from '@react-navigation/native-stack';
import React from 'react';
import {Image} from 'react-native';
import Loginpage from '../screen/Login/Login';
import SnsLogin from '../screen/Login/SnsLogin';
import LocalLogin from '../screen/Login/LocalLogin';
const NativeStack = createNativeStackNavigator();

const Login = () => (
  <NativeStack.Navigator
    screenOptions={{
      headerTitle: () => (
        <Image
          source={require('../assets/img/Small_Logo.png')}
          style={{width: 30, height: 30}}
        />
      ),
      headerShadowVisible: false,
      contentStyle: {backgroundColor: 'white'},
      headerTitleAlign: 'center',
    }}>
    <NativeStack.Screen name="login" component={Loginpage} />
    <NativeStack.Screen name="SnsLogin" component={SnsLogin} />
    <NativeStack.Screen name="LocalLogin" component={LocalLogin} />
  </NativeStack.Navigator>
);

export default Login;
