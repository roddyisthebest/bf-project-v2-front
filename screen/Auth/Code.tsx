import React, {useCallback, useEffect, useRef, useState} from 'react';
import styled from 'styled-components/native';
import {TextInput, Platform, Text} from 'react-native';
import {CommonActions} from '@react-navigation/native';
import {authCode} from '../../api/user';
import {api} from '../../api';
import Config from 'react-native-config';
import messaging from '@react-native-firebase/messaging';
import {NavigationProp, useNavigation} from '@react-navigation/native';

import {LoggedInParamList} from '../../navigation/Root';

const Container = styled.View`
  flex: 1;
  align-items: center;
  padding: 0 20px;
`;

const Title = styled.Text`
  font-size: 25px;
  font-weight: 800;
  color: black;
  margin-top: 20px;
`;

const SubTitle = styled.Text`
  font-size: 15px;
  font-weight: 400;
  color: #687684;
`;

const CellWrapper = styled.View`
  margin: 40px 0;
  flex-direction: row;
`;

const Cell = styled.TouchableOpacity`
  width: 40px;
  height: 40px;
  background-color: white;
  box-shadow: ${Platform.OS === 'android'
    ? '0 0 0 black '
    : '0px 0px 4px #00000054'};

  align-items: center;
  justify-content: center;
  font-size: 20px;
  border-radius: 10px;
`;

const Block = styled.View`
  width: 1px;
  height: 20px;
  background-color: blue;
`;

const Dot = styled.View`
  width: 10px;
  height: 10px;
  background-color: #10ddc2;
  border-radius: 10px;
`;

const Btn = styled.Pressable<{bkgColor: string}>`
  width: 95px;
  height: 35px;
  align-items: center;
  justify-content: center;
  border-radius: 40px;
  background-color: ${props => props.bkgColor};
`;

const BtnText = styled.Text<{color: string}>`
  font-size: 15px;
  color: ${props => props.color};
  font-weight: 700;
`;
const Code = () => {
  const navigation = useNavigation<NavigationProp<LoggedInParamList>>();

  const [val, setVal] = useState('');
  const input = useRef<any>();

  useEffect(() => {
    if (val.length === 6) {
      input.current.blur();
    }
  }, [val]);

  const focus = useCallback(() => {
    input.current.focus();
  }, []);

  async function getToken() {
    try {
      if (!messaging().isDeviceRegisteredForRemoteMessages) {
        await messaging().registerDeviceForRemoteMessages();
      }
      const phoneToken = await messaging().getToken();
      console.log('phone token', phoneToken);
      return api.post(`${Config.API_URL}/user/phonetoken`, {phoneToken});
    } catch (error) {
      console.error(error);
    }
  }
  const submit = useCallback(async () => {
    try {
      await authCode(val);
      getToken();
      return navigation.dispatch(
        CommonActions.reset({
          index: 0,
          routes: [{name: 'Setting'}],
        }),
      );
    } catch (e) {
      console.log(e);
    }
  }, [navigation, val]);

  return (
    <Container>
      <Title>6자리 코드입력</Title>
      <SubTitle style={{marginTop: 20}}>
        This app provides private content.
      </SubTitle>
      <SubTitle>Please enter a 6-digit code to prove yourself.</SubTitle>
      <CellWrapper>
        <Cell style={{marginRight: 5, elevation: 8}} onPress={focus}>
          {val.length === 0 ? <Block /> : val[0] ? <Dot /> : null}
        </Cell>
        <Cell style={{marginHorizontal: 5, elevation: 8}} onPress={focus}>
          {val.length === 1 ? <Block /> : val[1] ? <Dot /> : null}
        </Cell>
        <Cell style={{marginHorizontal: 5, elevation: 8}} onPress={focus}>
          {val.length === 2 ? <Block /> : val[2] ? <Dot /> : null}
        </Cell>
        <Cell style={{marginHorizontal: 5, elevation: 8}} onPress={focus}>
          {val.length === 3 ? <Block /> : val[3] ? <Dot /> : null}
        </Cell>
        <Cell style={{marginHorizontal: 5, elevation: 8}} onPress={focus}>
          {val.length === 4 ? <Block /> : val[4] ? <Dot /> : null}
        </Cell>
        <Cell style={{marginLeft: 5, elevation: 8}} onPress={focus}>
          {val.length === 5 ? <Block /> : val[5] ? <Dot /> : null}
        </Cell>
        <TextInput
          keyboardType="number-pad"
          ref={input}
          style={{opacity: 0, position: 'absolute'}}
          value={val}
          onChangeText={(text: string) => {
            if (text.length < 7) {
              return setVal(text);
            }
            input.current.blur();
          }}
          returnKeyType="go"
          autoFocus
          testID="input"
        />
      </CellWrapper>
      <Btn
        bkgColor={val.length === 6 ? '#10DDC2' : 'lightgray'}
        onPress={submit}
        disabled={val.length !== 6}
        testID="button">
        <BtnText color={val.length === 6 ? 'white' : '#6f6f6f'}>VERIFY</BtnText>
      </Btn>
      <Text testID="inputValue" style={{display: 'none'}}>
        {val}
      </Text>
    </Container>
  );
};

export default Code;
