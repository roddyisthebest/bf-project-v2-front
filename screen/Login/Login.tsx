import React from 'react';
import styled from 'styled-components/native';
import {NavigationProp, useNavigation} from '@react-navigation/native';
import {NotLoggedInParamList} from '../../navigation/Root';

const Container = styled.View`
  align-items: center;
  flex: 1;
  padding: 0 30px;
`;

const LoginBtn = styled.TouchableOpacity`
  width: 100%;
  height: 70px;
  border: 1px solid #cdcdcd;
  border-radius: 100px;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;

const LocalLoginBtn = styled.TouchableOpacity`
  width: 100%;
  height: 70px;
  border-radius: 100px;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background-color: black;
  margin: 10px 0;
`;

const LocalLoginText = styled.Text`
  color: white;
  font-weight: 700;
  font-size: 25px;
  margin-left: 15px;
`;

const LoginText = styled.Text`
  color: black;
  font-weight: 700;
  font-size: 25px;
  margin-left: 15px;
`;
const LoginBkg = styled.Image`
  width: 25px;
  height: 25px;
`;

const Login = () => {
  const navigation = useNavigation<NavigationProp<NotLoggedInParamList>>();
  return (
    <Container>
      {/* <Input placeholder="아이디" placeholderTextColor="#000" />
      <Input
        placeholder="비밀번호"
        secureTextEntry={true}
        placeholderTextColor="#000"
      /> */}
      <LocalLoginBtn
        onPress={() => {
          navigation.navigate('LocalLogin', {});
        }}>
        <LocalLoginText>로그인</LocalLoginText>
      </LocalLoginBtn>

      <LoginBtn
        onPress={() => {
          navigation.navigate('SnsLogin', {});
        }}>
        <LoginBkg source={require('../../assets/img/Kakao.png')} />
        <LoginText>KAKAO 로그인</LoginText>
      </LoginBtn>
    </Container>
  );
};

export default Login;
