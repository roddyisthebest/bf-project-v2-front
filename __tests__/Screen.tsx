import 'react-native';
import React from 'react';
import Code from '../screen/Auth/Code';
import {render, screen, fireEvent} from '@testing-library/react-native';
import axios from 'axios';
import Setting from '../screen/Auth/Setting';
import Login from '../screen/Login/Login';
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter');
jest.mock('react-redux');
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
  };
});
const accessToken: string =
  'ZtMM-4xqBgO0DoUjoFYtHx9nXe_F84kgStggI61aCj1ylwAAAYMbJr1_';

describe('Code', () => {
  const code: string = '060419';
  it('코드 값 변경 ', async () => {
    render(<Code />);
    fireEvent.changeText(screen.getByTestId('input'), code);
    const codeOutput = await screen.findByTestId('inputValue');
    expect(codeOutput).toHaveTextContent(code);
  });
  it('api 통신', async () => {
    try {
      const {status} = await axios.post(
        'https://api.bf-church.click/user/auth/code',
        {code},
        {
          headers: {
            Authorization: `bearer ${accessToken}`,
          },
        },
      );
      expect(status).toBe(200);
    } catch (e) {
      expect(e).toBeNaN();
    }
  });
  it('스냅샷', () => {
    render(<Code />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});

describe('Setting', () => {
  it('스위치 값 변경', async () => {
    render(<Setting />);
    let switchButton = screen.getByTestId('button');
    fireEvent(switchButton, 'onValueChange', true);
    const switchOutput = await screen.findByTestId('inputValue');
    expect(switchOutput).toHaveTextContent('true');
  });

  it('api 통신', async () => {
    try {
      const {status} = await axios.put(
        'https://api.bf-church.click/user/service',
        {
          tweet: true,
          pray: true,
          penalty: true,
        },
        {
          headers: {
            Authorization: `bearer ${accessToken}`,
          },
        },
      );
      expect(status).toBe(200);
    } catch (e) {
      expect(e).toBeNaN();
    }
  });
  it('스냅샷', () => {
    render(<Setting />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});

describe('Login', () => {
  it('스냅샷', () => {
    render(<Login />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
