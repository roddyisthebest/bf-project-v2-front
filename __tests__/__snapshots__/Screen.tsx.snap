// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code 스냅샷 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 0,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 0,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 25,
          "fontWeight": "800",
          "marginTop": 20,
        },
      ]
    }
  >
    6자리 코드입력
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#687684",
          "fontSize": 15,
          "fontWeight": "400",
        },
        Object {
          "marginTop": 20,
        },
      ]
    }
  >
    This app provides private content.
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#687684",
          "fontSize": 15,
          "fontWeight": "400",
        },
      ]
    }
  >
    Please enter a 6-digit code to prove yourself.
  </Text>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginBottom": 40,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 40,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "elevation": 8,
          "fontSize": 20,
          "height": 40,
          "justifyContent": "center",
          "marginRight": 5,
          "opacity": 1,
          "shadowColor": "#00000054",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 4,
          "width": 40,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "blue",
              "height": 20,
              "width": 1,
            },
          ]
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "elevation": 8,
          "fontSize": 20,
          "height": 40,
          "justifyContent": "center",
          "marginHorizontal": 5,
          "opacity": 1,
          "shadowColor": "#00000054",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 4,
          "width": 40,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "elevation": 8,
          "fontSize": 20,
          "height": 40,
          "justifyContent": "center",
          "marginHorizontal": 5,
          "opacity": 1,
          "shadowColor": "#00000054",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 4,
          "width": 40,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "elevation": 8,
          "fontSize": 20,
          "height": 40,
          "justifyContent": "center",
          "marginHorizontal": 5,
          "opacity": 1,
          "shadowColor": "#00000054",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 4,
          "width": 40,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "elevation": 8,
          "fontSize": 20,
          "height": 40,
          "justifyContent": "center",
          "marginHorizontal": 5,
          "opacity": 1,
          "shadowColor": "#00000054",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 4,
          "width": 40,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "elevation": 8,
          "fontSize": 20,
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 5,
          "opacity": 1,
          "shadowColor": "#00000054",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 4,
          "width": 40,
        }
      }
    />
    <TextInput
      autoFocus={true}
      keyboardType="number-pad"
      onChangeText={[Function]}
      returnKeyType="go"
      style={
        Object {
          "opacity": 0,
          "position": "absolute",
        }
      }
      testID="input"
      value=""
    />
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    bkgColor="lightgray"
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "lightgray",
          "borderRadius": 40,
          "height": 35,
          "justifyContent": "center",
          "width": 95,
        },
      ]
    }
    testID="button"
  >
    <Text
      color="#6f6f6f"
      style={
        Array [
          Object {
            "color": "#6f6f6f",
            "fontSize": 15,
            "fontWeight": "700",
          },
        ]
      }
    >
      VERIFY
    </Text>
  </View>
  <Text
    style={
      Object {
        "display": "none",
      }
    }
    testID="inputValue"
  >
    
  </Text>
</View>
`;

exports[`Login 스냅샷 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 0,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 0,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 27,
          "fontWeight": "900",
          "marginBottom": 50,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 35,
        },
      ]
    }
  >
    큰숲 청년부에서 무슨 일이 
    

    일어나고 있는지 알아보세요.
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#cdcdcd",
        "borderRadius": 100,
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 70,
        "justifyContent": "center",
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/img/Kakao.png",
        }
      }
      style={
        Array [
          Object {
            "height": 25,
            "width": 25,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 25,
            "fontWeight": "800",
            "marginLeft": 15,
          },
        ]
      }
    >
      KAKAO 로그인
    </Text>
  </View>
</View>
`;

exports[`Setting 스냅샷 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 25,
          "fontWeight": "800",
          "marginTop": 20,
        },
      ]
    }
  >
    기능 사용 설정
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#687684",
          "fontSize": 15,
          "fontWeight": "400",
        },
        Object {
          "marginTop": 20,
        },
      ]
    }
  >
    Welcome to roddyisthebest's world 🔥
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#687684",
          "fontSize": 15,
          "fontWeight": "400",
        },
      ]
    }
  >
    Please select the function you want to use.
  </Text>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#ced5dc",
          "borderBottomWidth": 0.5,
          "paddingBottom": 25,
          "paddingLeft": 40,
          "paddingRight": 40,
          "paddingTop": 25,
          "width": "100%",
        },
        Object {
          "borderTopColor": "#ced5dc",
          "borderTopWidth": 0.5,
          "marginTop": 25,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 10,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "800",
            },
          ]
        }
      >
        매일성경
      </Text>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 31,
            "width": 51,
          }
        }
        value={false}
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#687684",
            "fontSize": 12,
            "fontWeight": "400",
          },
        ]
      }
    >
      매일성경(큐티) 업로드를 사용할 수 있습니다.
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#ced5dc",
          "borderBottomWidth": 0.5,
          "paddingBottom": 25,
          "paddingLeft": 40,
          "paddingRight": 40,
          "paddingTop": 25,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 10,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "800",
            },
          ]
        }
      >
        기도제목
      </Text>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 31,
            "width": 51,
          }
        }
        value={false}
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#687684",
            "fontSize": 12,
            "fontWeight": "400",
          },
        ]
      }
    >
      기도제목을 업로드하고 수정 및 삭제 할 수 있습니다.
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#ced5dc",
          "borderBottomWidth": 0.5,
          "paddingBottom": 25,
          "paddingLeft": 40,
          "paddingRight": 40,
          "paddingTop": 25,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 10,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "800",
            },
          ]
        }
      >
        벌금
      </Text>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 31,
            "width": 51,
          }
        }
        testID="button"
        value={false}
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#687684",
            "fontSize": 12,
            "fontWeight": "400",
          },
        ]
      }
    >
      매일 성경(큐티) 벌금 로직에 참여할 수 있습니다.
    </Text>
  </View>
  <View
    accessible={true}
    bkgColor="#10DDC2"
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#10DDC2",
          "borderRadius": 40,
          "height": 35,
          "justifyContent": "center",
          "marginTop": 30,
          "width": 95,
        },
      ]
    }
  >
    <Text
      color="white"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 15,
            "fontWeight": "700",
          },
        ]
      }
    >
      확인
    </Text>
  </View>
  <Text
    style={
      Object {
        "display": "none",
      }
    }
    testID="inputValue"
  >
    false
  </Text>
</View>
`;
